src/modules/person/api/person.controller.ts

```diff
@@ -441,6 +444,30 @@ export class PersonController {
         return new PersonLockResponse(`User has been successfully ${lockUserBodyParams.lock ? '' : 'un'}locked.`);
     }
 
+    @Post(':personId/sync')
+    @HttpCode(HttpStatus.OK)
+    @ApiOkResponse({ description: 'User will be synced.' })
+    @ApiNotFoundResponse({ description: 'The person was not found.' })
+    @ApiForbiddenResponse({ description: 'Insufficient permissions to perform operation.' })
+    @ApiInternalServerErrorResponse({ description: 'An internal server error occurred.' })
+    @ApiBadGatewayResponse({ description: 'A downstream server returned an error.' })
+    public async syncPerson(
+        @Param('personId') personId: string,
+        @Permissions() permissions: PersonPermissions,
+    ): Promise<void> {
+        const personResult: Result<Person<true>> = await this.personRepository.getPersonIfAllowed(
+            personId,
+            permissions,
+            // TODO SPSH-1136 Update the RollenSystemRecht
+            [RollenSystemRecht.PERSONEN_VERWALTEN],
+        );
+        if (!personResult.ok) {
+            throw new NotFoundOrNoPermissionError(personId);
+        }
+
+        this.eventService.publish(new PersonExternalSystemsSyncEvent(personId));
```

Comment:
As far as I know, events should be published in the domain/persistence instead of api layer? Is it good practice for the eventService to be injected in a controller?

Reply:
Going to talk with an architect

Reply:
Should not create an extra layer just for the event